src/utils/index.js

```diff
@@ -0,0 +1,137 @@
+const { BigNumber } = require('bignumber.js');
+const safeAbi = require('../abis/SafeAbi.json');
+const { zeroAddress } = require('./constants');
+
+const baseGasValue = (hexValue) => {
+  switch (hexValue) {
+  case '0x': return 0;
+  case '00': return 4;
+  default: return 68;
+  }
+};
+
+const estimateBaseGasCosts = (dataString) => {
+  // eslint-disable-next-line
+  const reducer = (accumulator, currentValue) => accumulator += baseGasValue(currentValue);
+  return dataString.match(/.{2}/g).reduce(reducer, 0);
+};
+
+const estimateBaseGas = async (
+  cpkProvider,
+  to,
+  value,
+  data,
+  operation,
+  safeTxGas,
+  gasToken,
+  refundReceiver,
+  signatureCount,
+  nonce,
+  gasPrice,
+) => {
+  // (array count (3 -> r, s, v) + ecrecover costs) * signature count
+  const signatureCost = signatureCount * (68 + 2176 + 2176 + 6000);
```

Comment:
this is actually not important here, check https://github.com/gnosis/safe-react/pull/752/files